R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,293 @@
+
+    # -----------------------------------------------------------------------------
+    # 2.perform computations
+    L132.GWh_province_F_elec_out <- CPSY_GWh_province_F_elec_out %>%
+      filter(year %in% HISTORICAL_YEARS_c) %>%
+      subset(select = c(-other, -row)) %>%
+      map_province_name(province_names_mappings, "province", TRUE) %>%
+      # In Tibet, electricity in 'other' will be split between oil, gas, solar, and geothermal in 2010. For prior years
+      # it will be moved into coal.
+      mutate(oil = 0,
+             natural.gas = 0,
+             oil = replace(oil, province == "XZ", filter(CPSY_GWh_province_F_elec_out, province.name == "Tibet" & year == 2010)$other / 4),
+             natural.gas = replace(oil, province == "XZ", filter(CPSY_GWh_province_F_elec_out, province.name == "Tibet" & year == 2010)$other / 4),
+             coal = coal + coal.adj,
+             wind = wind + wind.adj,
+             nuclear = nuclear + nuc.adj,
+             solar = solar + solar.adj) %>%
+      subset(select = c(-coal.adj, -wind.adj, -nuc.adj, -solar.adj)) %>%
+      mutate(biomass = coal) %>%
+      gather(fuel, value, c(-province, -year)) %>%
+      filter(value == 0 | value > 0.1) %>%
+      complete(nesting(province, fuel), year = HISTORICAL_YEARS_c) %>%
+      arrange(province, fuel, year) %>%
+      group_by(province, fuel) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup %>%
+      mutate(fuel = as.character(fuel)) ->
+      L132.GWh_province_F_elec_out
+
+    # Oil and gas does not exist in the CPSY so we will use the CESY for those
+    # Note these are in terms of energy in.  Since we are assuming constant efficiencies for all provinces
+    # mixing these with the elec_out will be fine for approtioning to provinces
+    L101.inNBS_Mtce_province_S_F %>%
+      filter(sector == "electricity" & fuel %in% c("refined liquids", "gas")) ->
+      L132.Mtce_province_F_elec_in
+
+    L132.Mtce_province_F_elec_in[is.na(L132.Mtce_province_F_elec_in)] <- 0
+    L132.Mtce_province_F_elec_in %>%
+      subset(select = -sector) %>%
+      bind_rows(L132.GWh_province_F_elec_out) ->
+      L132.Mix_province_F_elec
+
+    L132.Mix_province_F_elec %>%
+      # In Tibet, refined liquids inherits oil and gas inherits natural gas
+      mutate(fuel = replace(fuel, fuel == "oil" & province == "XZ", "refined liquids"),
+             fuel = replace(fuel, fuel == "natural.gas" & province == "XZ", "gas"),
+             # In Tibet our oil and gas numbers are in electricity out and they need to be energy in in Mtce. This is a total hack but
+             # just divide by the efficiency. Gas is 0.55 and oil is about 0.35
+             tag = 0,
+             tag = replace(tag, fuel == "refined liquids" & province == "XZ", 1),
+             tag = replace(tag, fuel == "gas" & province == "XZ", 2),
+             value = replace(value, tag != 0, 0),
```

Comment:
`tag` is unnecessary: `value = replace(value, province == "XZ" & fuel %in% c("refined liquids", "gas"), 0),`